packages/router/src/router.tsx

```diff
@@ -178,7 +180,11 @@ export default function useMatch(
 					if ( typeof value === 'function' ) {
 						return [
 							key,
-							value( { query, params: result.params } ),
+							value( {
+								query,
+								params: result.params,
+								context,
```

Comment:
```suggestion
								...context,
```
This goes along with my suggestion to rename `context`. This way resolver functions also can access more specifically named properties.